conf/app.routes

```diff
@@ -1,3 +1,3 @@
 # microservice specific routes
 
-GET        /hello-world             uk.gov.hmrc.transitmovementsvalidator.controllers.MicroserviceHelloWorldController.hello
+POST        /messages/:messageType/validation             uk.gov.hmrc.transitmovementsvalidator.controllers.MessagesController.validate(messageType)
```

Comment:
This isn't as RPC as I worried, but this route signature still seems a bit naff.

I know we send an XML body of a message, but we don't create a message as a result. If we create anything we create a ValidationResult of some sort.

@crystalmark what do you think?

Reply:
@crystalmark was the one that suggested that URL!

I was a little on the edge about whether or not we should be returning anything like a `ValidationResult`. We don't create anything - we validate and either it's OK (so nothing to return), or it's not (opted for BadRequest and creating that Schema error object).

If we _do_ want to return a `ValidationResult`, what would a successful result contain? All I could think of was a `success: true` and maybe return the message type, but that's what HTTP 200 and knowledge of what you requested will tell you.

Reply:
Ah, fair enough. Tim is the REST-King, if he's happy I'm happy.

but yeah, it depends on what we'd *want*, you're right. Other than maybe wanting to keep a validation record against the movement, I'm not sure what it would contain outside of timestamps for received & processed. Could be useful info for long-term performance if it was logged, but for any of our services.

I'm happy to consider it resolved if you are.

Reply:
Let's see what Tim thinks, I wouldn't mind an extra thought or two on this.

Reply:
I see returning 200 for either valid or invalid XML as correct.  This service is all about validating the XML - the ValidationResult will contain a list of validation errors and if the list is empty, it passed validation.  If it was RPC, it could be named /getValidationErrorList or something.
If we can't parse the XML (e.g. if they send JSON or an empty payload) then it's a BadRequest.
Still open for discussion though... :-)

Reply:
That's actually fair - 200 should indicate that the _action_ of validating was successful (which is what we requested), even if it failed validation (it's a successful result).

I can get behind that.